### PR TITLE
Hack to enable '=' in the middle of command arguments.

### PR DESCRIPTION
**Pull request by tiago4orion**
WIP: PoC to verify alternatives to syntax.

The problem is that '=' cannot be used in command arguments without quoting to avoid the ambiguity below (see issue #114 ):

    ls = "a"

It's a variable assignment? or a 'ls' listing on files '=' and "a"?

Current syntax is very straightforward, you must be explicit about what you're writing. Then, if you want execute listing on files '=' and 'a', you must quote the '='.

    ls "=" a
or
    ls "=" "a"

Otherwise, it's an assignment.

Ok, then?

This commit leaves the problem above as is, but enable '=' in other cases, like below:

    make collection=project1

or

    docker run -e VAR1= -e VAR2= something

But EQUAL is still *prohibited* in the cases below:

    1. make otheroption =something # (solvable)
    2. make otheroption = othervalue # (solvable) 
    3. make = something # (solvable, but only because assignment require quotes in the rhs)
    4. make = "something" # (impossible, this is an assignment by spec...)

1 & 2 are easy.
Solving 3 will make the parser code hard to follow or will require changes in the lexer to emit context to each token.

Maybe this confuses the syntax, and people could be more *surprise* of this behaviour than the enforcement of quoting '=' ... I don't know.

@katcipis @lborguetti @patito What do you guys think? It's a good direction?

### CODE REVIEW

**Comment by tiago4orion** [comment]
Addendum:

Currently, the commands below are valid ones:

```sh
./configure --prefix=/usr
go tool cover -html=c.out
```

The reason is that the ambiguity exists only when using '=' with token.Ident. In the cases above, "--prefix" and "-html" are tokens of type token.Arg.

**Comment by katcipis** [comment]
For me the cases that are already covered are good enough :-)

**Comment by tiago4orion** [comment]
OK.
